### DISCUSSION TITLE
Eslint implementation in the AWS Lambda application

### DISCUSSION THREAD

**Discussion by sulemanelahi** [eslint/eslint#16689] [Q&A]
I am using AWS lambda dependencies layer which is a child folder that includes (node_modules and .eslintrc.json).

Project structure:
```.
├── dependencies
|   └── nodejs
|       ├── node_modules
|       ├── package.json
|       └── .eslintrc.json  
└── src
    └── signup
    |   ├── signup.ts
    |   └── midllewares
    |       └── signup-validation.ts
    └── login
        ├── login.ts
        └── midllewares
            └── login-validation.ts
```
My problem is that eslint rules don't work on typescript inside src directory, it only works inside nodejs directory.

How to apply the rules on all .eslintrc.json's parent folders.

The solution I tried is to do npm install in the project root and move .eslintrc.json there, but it's an ugly solution to have package.json twice

Here is the content of .eslintrc.json
```
{
    "env": {
        "browser": true,
        "es2021": true
    },
    "extends": [
        "eslint:recommended",
        "plugin:@typescript-eslint/recommended"
    ],
    "overrides": [
    ],
    "parser": "@typescript-eslint/parser",
    "parserOptions": {
        "ecmaVersion": "latest"
    },
    "plugins": [
        "@typescript-eslint"
    ],
    "rules": {
			"no-unused-vars": "error"
    }
}

```

**Comment by sam3k** [comment]
Hi @sulemanelahi, seems to me like you are choosing to use the `.eslintrc.json` from a vendor a third party library. My suggestion would be to:

1. Create your own `.eslintrc.json`
2. Move it to the root of your application
3. Add a `.eslintignore` to the root of your application
4. Inside `.eslintignore` add `dependencies/nodejs/node_modules` so your `.eslintrc.json` does not override the `.eslintrc.json` in `dependencies/nodejs/.eslinterc.json`

```
├─ .eslintrc.json // <-- Add your own here
├─ .eslintignore // <-- also add a eslintignore and ignore anything inside dependencies/nodejs/*
├── dependencies
|   └── nodejs
|       ├── node_modules
|       ├── package.json
|       └── .eslintrc.json  
└── src
    └── signup
    |   ├── signup.ts
    |   └── midllewares
    |       └── signup-validation.ts
    └── login
        ├── login.ts
        └── midllewares
            └── login-validation.ts
```

Not sure if this is what you are looking for. Hope it helps.

**Reply by sulemanelahi** [reply]
Thanks for your reply, @sam3k 
The Eslint configuration file expects `node_modules` to be in the same folder as `.eslintrc.json` when we run the `eslint` command.

However, I found out that we could write eslint configuration inside `package.json`, which technically puts it in the same directory as `node_modules`. 

```
"scripts": {
  "lint": "eslint --config ./package.json ../../src/**/*.ts"
},
"eslintConfig": {
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
      "project": "../../tsconfig.json"
  },
  "plugins": [
      "@typescript-eslint"
  ],
  "extends": [
      "eslint:recommended",
      "plugin:@typescript-eslint/recommended"
  ],
  "rules": {
      "@typescript-eslint/no-unused-vars": "error"
  }
}
```
This solved the problem

**Reply by sam3k** [reply]
Glad you were able to figure it out!
